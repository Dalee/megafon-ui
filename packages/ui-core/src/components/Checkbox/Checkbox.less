@import '~styles/base.less';

@b: e('.mfui-checkbox');

@{b} {
    margin: 0 5px 12px 5px;

    &__inner {
        position: relative;
        display: inline-block;
    }
    &__label {
        display: inline;
        cursor: pointer;
    }

    &__input {
        opacity: 0;
        position: absolute;
        left: 0;
        top: 0;
        z-index: -1;
        width: 0;
        height: 0;
    }
    &__custom-input {
        position: absolute;
        left: 0;
        box-sizing: border-box;
        display: flex;
        align-items: center;
        justify-content: center;
        width: 20px;
        height: 20px;
        border: 2px solid;
        border-radius: 2px;
    }
    &__icon {
        width: 32px;
        height: 32px;
        display: block;
        user-select: none;
        opacity: 0;
    }
    &__extra-content {
        position: relative;
        left: 8px;
        margin-right: -20px;
        display: inline-block;
        vertical-align: middle;
        margin-top: -2px;
    }

    &_checked {
        @{b} {
            &__custom-input {
                border: none;
            }
            &__icon {
                opacity: 1;
            }
        }
    }

    &_font-size_regular {
        font-size: 15px;
        line-height: 24px;

        @{b} {
            &__inner {
                padding: 4px 15px 4px 35px;
            }
            &__custom-input {
                top: 6px;
            }
        }
    }

    &_font-size_small {
        font-size: 12px;
        line-height: 16px;
        @{b} {
            &__inner {
                padding: 1px 15px 4px 35px;
            }
            &__custom-input {
                top: 0;
            }
        }
    }

    &_color_dark {
        @{b} {
            &__custom-input {
                border-color: @spbSky2;
            }
            &__label_no-touch:hover @{b}__custom-input {
                border-color: @greenHover;
            }
            &__label:active @{b}__custom-input {
                border-color: @down !important;
                background-color: @down;
            }
            &__icon {
                fill: @clearWhite;
            }
        }
    }

    &_color_dark&_checked {
        @{b} {
            &__custom-input {
                border-color: @green;
                background-color: @green;
                color: @white;
            }
            &__label_no-touch:hover @{b}__custom-input {
                background-color: @greenHover;
            }
            &__label:active @{b}__custom-input {
                background-color: @down !important;
            }
        }
    }

    &_color_dark&_checked&_disabled {
        @{b} {
            &__custom-input,
            &__label:active @{b}__custom-input {
                background-color: @spbSky2;
            }
        }
    }

    &_color_dark&_disabled {
        @{b} {
            &__label {
                cursor: default;
                pointer-events: none;
            }
            &__custom-input,
            &__label:active @{b}__custom-input {
                border-color: @spbSky2 !important;
                background-color: @spbSky1;
            }
        }
    }

    &_color_dark&_error {
        @{b} {
            &__custom-input {
                border-color: @fury;
            }
        }
    }

    &_color_light {
        color: @clearWhite;
        @{b} {
            &__custom-input {
                color: @green;
                border-color: @clearWhite;
                background-color: @clearWhite;
            }
        }
    }
}
