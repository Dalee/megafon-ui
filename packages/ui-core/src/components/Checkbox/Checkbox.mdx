---
name: Checkbox
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Checkbox from './Checkbox';
import Info from 'icons/System/16/Info_16.svg';

export const LightColorWrapper = ({ children }) => {
    return (
        <div style={{ backgroundColor: '#00b956', padding: '20px 10px 8px 10px' }}>
            {children}
        </div>
    );
}

# Checkbox

## Properties

<PropsTable of={Checkbox} />

export class CheckboxWrapper extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            checked: false,
        }
    }
    handleChange = () => {
        this.setState(prevState => ({ checked: !prevState.checked }))
    }
    render() {
        return this.props.children({
            checked: this.state.checked,
            onChange: this.handleChange
        })
    }
}

## Basic usage

<Playground>
    <CheckboxWrapper>
        {({ checked, onChange }) =>
            <Checkbox checked={checked} onChange={onChange}>
                I agree
            </Checkbox>
        }
    </CheckboxWrapper>
</Playground>

## Checkbox usage with error

<Playground>
    <CheckboxWrapper>
        {({ checked, onChange }) =>
            <Checkbox checked={checked} onChange={onChange} error>
                I agree
            </Checkbox>
        }
    </CheckboxWrapper>
</Playground>

## Checkbox usage for light color scheme and small size

<Playground wrapper={LightColorWrapper}>
    <CheckboxWrapper>
        {({ checked, onChange }) =>
            <Checkbox
                checked={checked}
                onChange={onChange}
                color="light"
                fontSize="small"
            >
                I agree
            </Checkbox>
        }
    </CheckboxWrapper>
</Playground>

## Checkbox with extraContent

<Playground>
    <CheckboxWrapper>
        {({ checked, onChange }) =>
            <Checkbox
                checked={checked}
                onChange={onChange}
                extraContent={
                    <div style={{ backgroundColor: '#00B956', color: '#FFF', padding: '5px'}}>
                        Some Extra Content
                    </div>
                }
            >
                I agree
            </Checkbox>
        }
    </CheckboxWrapper>
</Playground>

## Disabled unchecked checkbox

<Playground>
    <CheckboxWrapper>
        {() => <Checkbox disabled>I agree</Checkbox>}
    </CheckboxWrapper>
</Playground>

## Disabled checked checkbox

<Playground>
    <CheckboxWrapper>
        {() => <Checkbox checked disabled>I agree</Checkbox>}
    </CheckboxWrapper>
</Playground>
