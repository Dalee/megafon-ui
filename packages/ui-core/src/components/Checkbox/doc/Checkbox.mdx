---
name: Checkbox
menu: Компоненты
route: components/checkbox
---

import { Playground, Props } from 'docz';
import Checkbox from '../Checkbox';
import Example from './Checkbox.example.mdx';

# Checkbox

<ImportExample name="Checkbox" />

<DoczTabs>
    <Example />
    <Props of={Checkbox} />
</DoczTabs>
