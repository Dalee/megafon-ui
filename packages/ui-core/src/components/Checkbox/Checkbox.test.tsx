import * as React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './Checkbox';

const props = {
    className: 'testClassName',
    name: 'testInputName',
    value: 'testInputValue',
    checked: true,
    disabled: true,
    error: true,
    extraContent: 'text',
};

describe('<Checkbox />', () => {
    describe('layout', () => {
        it('renders Checkbox', () => {
            const wrapper = shallow(<Checkbox>Тестовая строка</Checkbox>);
            expect(wrapper).toMatchSnapshot();
        });

        it('renders Checkbox with props', () => {
            const wrapper = shallow(
                <Checkbox
                    {...props}
                    fontSize="small"
                    color="light"
                    onChange={jest.fn()}
                >
                    Тестовая строка
                </Checkbox>
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('renders Checkbox on mobile resolution', () => {
            const onTouchStart = window.ontouchstart;
            window.ontouchstart = jest.fn();

            const wrapper = shallow(<Checkbox>Тестовая строка на мобильном разрешении</Checkbox>);
            expect(wrapper).toMatchSnapshot();

            window.ontouchstart = onTouchStart;
        });
    });

    describe('handleChange', () => {
        it('calls onChange', () => {
            console.log('window.ontouchstart', window.ontouchstart);
            const handleChange = jest.fn();
            const event = {
                target: { value: 'test value' },
            };

            const wrapper = shallow(<Checkbox onChange={handleChange}>Тестовая строка</Checkbox>);

            wrapper.find('input').simulate('change', event);
            expect(handleChange).toBeCalledWith(event);
        });
    });
});
