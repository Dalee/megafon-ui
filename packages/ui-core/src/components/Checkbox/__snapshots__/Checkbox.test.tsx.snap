// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> layout renders Checkbox 1`] = `
<div
  className="mfui-checkbox mfui-checkbox_color_dark"
>
  <div
    className="mfui-checkbox__inner"
  >
    <label
      className="mfui-checkbox__label mfui-checkbox__label_no-touch"
    >
      <input
        checked={false}
        className="mfui-checkbox__input"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="mfui-checkbox__custom-input"
      >
        <test-file-stub
          className="mfui-checkbox__icon"
        />
      </div>
      Тестовая строка
    </label>
  </div>
</div>
`;

exports[`<Checkbox /> layout renders Checkbox with props 1`] = `
<div
  className="mfui-checkbox mfui-checkbox_color_light mfui-checkbox_checked mfui-checkbox_disabled mfui-checkbox_error testClassName"
>
  <div
    className="mfui-checkbox__inner"
  >
    <label
      className="mfui-checkbox__label mfui-checkbox__label_no-touch"
    >
      <input
        checked={true}
        className="mfui-checkbox__input"
        disabled={true}
        name="testInputName"
        onChange={[Function]}
        type="checkbox"
        value="testInputValue"
      />
      <div
        className="mfui-checkbox__custom-input"
      >
        <test-file-stub
          className="mfui-checkbox__icon"
        />
      </div>
      Тестовая строка
    </label>
    <span
      className="mfui-checkbox__extra-content"
    >
      text
    </span>
  </div>
</div>
`;
